packages/std/src/math.rs

```diff
@@ -0,0 +1,255 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::convert::{TryFrom, TryInto};
+use std::{fmt, ops};
+
+use crate::errors::{generic_err, underflow, StdError, StdResult};
+
+/// Decimal9 represents a fixed-point decimal value with 9 fractional digits.
+/// That is Decimal9(1_000_000_000) == 1
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Decimal9(pub u64);
+
+pub const DECIMAL_FRACTIONAL: u64 = 1_000_000_000;
+
+impl Decimal9 {
```

Comment:
This type could use a `pub fn zero() -> Decimal9 {` constructor and a `pub fn atomics(&self) -> u64 { self.0 }` getter.

Reply:
`Decimal(0)` is the same as `Decimal::zero()` would be. Do you see a real need for the function?

I see adding a function like `pub fn u64(&self) -> u64` would make it more readable than `self.0`, like with `Uint128`.

Reply:
If we keep `self.0` public, both for reading and for writing, we encourage the user to operate on atomic units rather than on the decimal value. `Decimal9::one()` is also the same as `Decimal9(1000000000)`. I think we should privatize the `Decimal9(xy)` constructor in favour of something explicit like `Decimal9::from_atomics(xy) // shooting myself in the foot explicitely`

Reply:
Okay, let's make it private and add the `zero()` method.
I don't think we need other constructors (yet) besides percent and permil - 1/1000 (we can add them later).

You are right, this is different than the case of Uint128.

We can just add a few math operators to it (also Ord, Eq) so that there should never been a need to look at the atomics.

Reply:
I just did some ugliness to figure out how to get a 1.5 ratio for test assertions.

```rust
let ratio = Decimal::one();
ratio.0 = ratio.0 * 3 / 2;
```

:grimacing: 

I'm sure you can make this better :crossed_fingers:

Reply:
3/2 is easy. But what about 2/3? This cannot be represented precisely as a decimal. Let's see ...

Reply:
`Decimal9::one() + Decimal9::percent(50)` seems solid and very little overhead

Reply:
Yup, adding Add is good